### PR TITLE
Use protoc from java/lib rather than Maven repository.

### PR DESCRIPTION
**Pull request by dlyongemallo**

### CODE REVIEW

File: java/lib/com/google/protobuf/protobuf-java/maven-metadata-local.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<metadata>
+  <groupId>com.google.protobuf</groupId>
+  <artifactId>protobuf-java</artifactId>
+  <versioning>
+    <release>2.6.1</release>
+    <versions>
+      <version>2.6.1</version>
+    </versions>
+    <lastUpdated>20151102180249</lastUpdated>
```
Comment (keghani) [inline java/lib/com/google/protobuf/protobuf-java/maven-metadata-local.xml]: What time zone is this supposed to be? UTC? I wonder why it's not in epoch seconds.

File: java/libphonenumber/pom.xml [hunk 2]
```diff
@@ -46,4 +46,18 @@
     </plugins>
   </build>
 
+  <repositories>
+    <repository>
+      <id>protobuf-java-lib</id>
+      <url>file://${project.basedir}/../lib</url>
+    </repository>
+  </repositories>
+  <dependencies>
```
Comment (keghani) [inline java/libphonenumber/pom.xml]: Thanks for moving this out of java/pom.xml :)

File: tools/java/common/pom.xml [hunk 3]
```diff
@@ -49,6 +49,12 @@
...
 
+  <repositories>
+    <repository>
+      <id>protobuf-java-lib</id>
+      <url>file://${project.basedir}/../../../java/lib</url>
+    </repository>
+  </repositories>
   <dependencies>
     <dependency>
       <groupId>junit</groupId>
```
Comment (keghani) [inline tools/java/common/pom.xml]: Fun fact: we have junit-4.8.1.jar under java/lib. I wonder what Maven is using, and who is using the one under java/lib.

**Comment by keghani** [comment]
Merging, but I still want to discuss above points.

File: java/lib/com/google/protobuf/protobuf-java/maven-metadata-local.xml (hunk 1 above)
Comment (dlyongemallo) [inline java/lib/com/google/protobuf/protobuf-java/maven-metadata-local.xml]: As discussed, this is used by maven to check against, e.g., ~/.m2/repository/com/google/protobuf/protobuf-java/2.6.1/protobuf-java-2.6.1.jar.lastUpdated - to determine whether a local version is newer or older than the one checked into our repo.

File: tools/java/common/pom.xml (hunk 3 above)
Comment (dlyongemallo) [inline tools/java/common/pom.xml]: There are some path issues with the junit that's checked in. We should fix that.

**Pull request merged**